### PR TITLE
[14.0][FIX] s_storage_type: allowed location storage types

### PR DESCRIPTION
**Pull request by sebalix**
Rewrite the `_compute_allowed_location_storage_type_ids` in a iterative
way (instead of using recursion) to avoid a RecursionError exception
when computing large recordsets.

NOTE: Still need some tests locally to ensure we get the same results.

### CODE REVIEW

**Comment by sebalix** [comment]
Tested locally on a customer db (~30k locations): I got the same results.

File: stock_storage_type/models/stock_location.py [hunk 1]
```diff
@@ -284,19 +284,31 @@ def _compute_location_is_empty(self):
...
         "location_storage_type_ids",
         "location_id",
-        "location_id.allowed_location_storage_type_ids",
+        "location_id.location_storage_type_ids",
     )
     def _compute_allowed_location_storage_type_ids(self):
-        for location in self:
+        # NOTE: this compute isn't recursive to avoid a RecursionError
+        # on huge recordsets.
+        # Process the root location first and finish with leafs
```
Comment (mmequignon) [inline stock_storage_type/models/stock_location.py]:
```suggestion
        # Process the root location first and finish with leaves
```

**Review by mmequignon — APPROVED** [review]
LG, tiny nitpick

**Review by TDu — APPROVED** [review]

**Pull request merged**